yarn-project/sequencer-client/src/sequencer/metrics.ts

```diff
@@ -140,6 +203,34 @@ export class SequencerMetrics {
     });
   }
 
+  observeSlotChange(slot: bigint | undefined, proposer: string) {
+    // sequencer went through the loop a second time. Noop
+    if (slot === this.lastSeenSlot) {
+      return;
+    }
+
+    if (typeof this.lastSeenSlot === 'bigint') {
+      this.missedSlots.add(1, {
+        [Attributes.BLOCK_PROPOSER]: proposer,
+      });
+    }
+
+    if (typeof slot === 'bigint') {
+      this.slots.add(1, {
+        [Attributes.BLOCK_PROPOSER]: proposer,
+      });
+    }
+
+    this.lastSeenSlot = slot;
+  }
```

Comment:
This function gets called on every iteration of the sequencer work loop and tries to determine if a slot was missed prior to the current run.